lua/packer.lua

```diff
@@ -796,6 +818,89 @@ packer.plugin_complete = function(lead, _, _)
   return completion_list
 end
 
+---Snapshots installed plugins
+---@param snapshot_name string @ can be an absolute path or just a filename
+packer.snapshot = function(snapshot_name)
+  local async = require('packer.async').sync
+  local await = require('packer.async').wait
+  local log = require_and_configure 'log'
+
+  async(function()
+    local snapshot = require 'packer.snapshot'
+    local snapshot_path = snapshot_name
+    manage_all_plugins()
+    local fmt = string.format
+    log.debug(fmt('Taking snapshots of currently installed plugins to %s...', snapshot_name))
+    if vim.fn.fnamemodify(snapshot_name, ":p") ~= snapshot_path then -- is not absolute path
+      snapshot_path = util.join_paths(config.snapshot_path, snapshot_path) -- save to default path
+
+    end
+    await(snapshot(snapshot_path, plugins))
+    local msg = fmt("Snapshot '%s' complete", snapshot_path)
+    log.info(msg)
+    print(msg)
+    packer.on_complete() --not sure if it should fire packer.on_complete()
+  end)()
+end
+
+---Deletes the snapshot provided
+---@param snapshot_name string @name of the snapshot or the absolute path
+---@return boolean @true if success, false if failed
+packer.delete = function (snapshot_name)
+  local async = require('packer.async').sync
+  local log = require_and_configure 'log'
+
+    async(function ()
+      local snapshot_path = snapshot_name
+      if vim.fn.fnamemodify(snapshot_path, ":p") ~= snapshot_path then
```

Comment:
We use this "absolute path or not" logic a few times - maybe it should be a utility function?